src/praisonai-agents/praisonaiagents/agent/tool_execution.py

```
@@ -1320,6 +1332,15 @@ async def _async_approval():
                 else:
                     return ApprovalDecision(approved=True, reason="Not a dangerous tool")
         else:
+            # No approval backend configured. An explicit PermissionManager
+            # ``ask`` rule must still gate the call, so mark it required in the
+            # approval registry (idempotent) before delegating so the registry
+            # prompts instead of silently allowing.
+            if manager_forces_approval:
+                try:
+                    get_approval_registry().add_requirement(tool_name)
+                except Exception:  # noqa: BLE001
+                    pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)